internal/kafka/internal/workers/clusters_mgr.go

```diff
@@ -659,7 +659,11 @@ func (c *ClusterManager) reconcileAddonOperator(provisionedCluster api.Cluster)
 
 	if strimziOperatorIsReady && kasFleetshardOperatorIsReady && (clusterLoggingOperatorIsReady || c.OCMConfig.ClusterLoggingOperatorAddonID == "") {
 		glog.V(5).Infof("Set cluster status to %s for cluster %s", api.ClusterWaitingForKasFleetShardOperator, provisionedCluster.ClusterID)
-		if err := c.ClusterService.UpdateStatusAndClientId(provisionedCluster, api.ClusterWaitingForKasFleetShardOperator, params.GetParam(services.KasFleetshardOperatorParamServiceAccountId)); err != nil {
+		if err := c.ClusterService.
+			UpdateStatusAndClient(provisionedCluster,
+				api.ClusterWaitingForKasFleetShardOperator,
+				params.GetParam(services.KasFleetshardOperatorParamServiceAccountId),
+				params.GetParam(services.KasFleetshardOperatorParamServiceAccountSecret)); err != nil {
 			return errors.Wrapf(err, "failed to update local cluster %s status: %s", provisionedCluster.ClusterID, err.Error())
 		}
```

Comment:
Let's also make sure that `ready` and `waiting_for_kas_fleetshard_operator` clusters are also updated:
- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/43ba6d92a9af9fcffedde181b37c66a0847e76cc/internal/kafka/internal/workers/clusters_mgr.go#L596-L601
That's where it is happening.

Reply:
Good catch! Should be fixed now